core-bundle/src/DependencyInjection/Compiler/RegisterFragmentsPass.php

```diff
@@ -66,6 +69,18 @@ protected function registerFragments(ContainerBuilder $container, string $tag):
         $preHandlers = [];
         $registry = $container->findDefinition('contao.fragment.registry');
         $command = $container->findDefinition('contao.command.debug_fragments');
+        $callback = $container->findDefinition(CustomTemplateOptionsListener::class);
+        $table = null;
+
+        switch ($tag) {
+            case ContentElementReference::TAG_NAME:
+                $table = 'tl_content';
+                break;
```

Comment:
Is this really the best/only way to get the table name? It it necessary to have the table name at all? @contao/developers /cc

Reply:
I am not sure this is the best approach anyway. Instead of registering those templates in the compiler pass, may be we should retrieve the fragment's service and provide a getter function for in the abstract fragment class for the custom template? Though this would then require the service to be instantiated every time you edit an element in the back end.

Reply:
IDK. @aschempp @Toflar WDYT?

@fritzmg Why do we need the table name at all? Every fragment has an identifier and a `FragmentConfig` object – can we not use this information instead of the table name?

Reply:
Yeah, may be there is a better way :). I just need some pointers on how to get to the information, as I am not all that familiar with the inner workings of fragments yet.